apps/prairielearn/src/ee/auth/saml/router.ts

```diff
@@ -88,6 +92,12 @@ router.post(
     }
 
     // Only perform validation if we aren't rendering the above test page.
+    //
+    // Support for pulling in email from an attribute was added after all initial
+    // attributes, so we can't yet require it to be present. In the future, once
+    // we've specified such an attribute for all institutions, we can assert that
+    // the email attribute mapping and the corresponding value are both present.
```

Comment:
Does this require an issue to avoid forgetting?

Reply:
I don't think so; this is low priority and a long way off.